src/Vlingo.Actors/World.cs

```diff
@@ -425,7 +425,17 @@ public void RegisterDynamic(string name, object dep)
         /// <returns></returns>
         public TDependency ResolveDynamic<TDependency>(string name)
         {
-            return (TDependency)this.dynamicDependencies[name];
+            if(!this.dynamicDependencies.TryGetValue(name, out object value))
+            {
+                return default!;
+            }
+
+            if(value is TDependency)
```

Comment:
Not sure if we wanted to include this scenario, but there's some more tests to go along with this check.

Reply:
`value as TDependency` won't work since `TDependency` can be byVal type.

Reply:
@jamesmh just returning `(TDependency)value` if `TryGetValue` Is successful would be enough

Reply:
Ok, great. Will do 👍